scripts/abi_artifact_generator/config/architectures.yml

```diff
@@ -50,6 +54,8 @@
       bl {{ function_name }}
     return_from_function: |
       ret
+    return_from_function_with_cleanup: |
+      ret
```

Comment:
You're not doing any cleanup here, similar below

Reply:
Yeah, it's my bad on choosing the template name. Maybe I should actually modify the template to detect if this value was provided and fall back onto `return_from_function` in the cases where it was not 🤔 

What's your stance?

Reply:
1. I think it's a mistake if we ever use `return_from_function_with_cleanup` for an architecture for which we don't want to support callee-cleaned up ABIs. Can we put an invalid instruction to prevent assembling it entirely?
2. Why are we hardcoding the amount of stuff to pop from the stack? `ret $8`

Reply:
1. Well yeah, this is only ever used for ABI for which calle-cleared macro is explicitly set. Yeah, an invalid instruction makes a lot of sense.
2. It's a pointer, which is a constant for a given architecture. If you prefer, I can give it a name, but it's an ugly hack either way - named or not.

Reply:
> It's a pointer, which is a constant for a given architecture. If you prefer, I can give it a name, but it's an ugly hack either way - named or not.

This means we can only test having a pointer-sized stack argument (for callee-saved ABIs)? Can't we leave a placeholder and fill it in later on?

Reply:
This is specifically for return value stuff. This instruction is only ever inserted when a function returns stuff by taking a pointer where to place it as a stack argument.

The common case with argument is, luckily, doesn't require us to do anything special since `setjmp`/`longjmp` pair handles it for us.

Reply:
Ah so this is for the wrapper function that pushed the argument?

Please add a comment at the top of the file what these snippets are for and document each field.

Reply:
More like "wrapped" function that readies the return value, but yeah, popping one register is all it needs.

Documentation - sounds good.

Reply:
The files started to look ugly, so I moved the documentation outside - into a dedicated markdown file. And while at it, I also added the rest of the config in as well - for completeness.

See `scripts/abi_artifact_generator/config/readme.md`.

Reply:
Here's the link straight to it: https://github.com/revng/revng-qa/tree/f0d5b2c21457434a1c441aa1f3b276782399d25d/scripts/abi_artifact_generator/config

Reply:
Do not link commits, I told you many times.
I'll continue from `readme.md`.